addons/sale_project/models/sale_order.py

```diff
@@ -300,3 +300,9 @@ def write(self, values):
     def _compute_completed_task_percentage(self):
         for so in self:
             so.completed_task_percentage = so.tasks_count and so.closed_task_count / so.tasks_count
+
+    def action_confirm(self):
+        if len(self) == 1 and self.env.context.get('create_for_project_id') and self.state == 'sale':
```

Comment:
the len(self) check does not looks useful to me.  In our use case, i fail to see a situation where we could have more than one so on the recordset and the context set.

Reply:
Actually, since the `create_for_project_id` context is added when we click on stat button `make billable`, we will always have only one record here. 🤷🏼‍♂️ 

We could maybe be smarter and manage multi recordset but it does not seem useful for me. 🤷🏼‍♂️ 

Also, if I remove `len(self) == 1` then `self.state == 'sale'` will crash. 🙈 

We could maybe do something like this if you prefer? 🙂 

```suggestion
        if self.env.context.get('create_for_project_id') and all(sol.state == 'sale' for sol in self):
```
What do you think? 🤷🏼‍♂️ 🙂